pkg/cmd/objects/operations/operations.go

```diff
@@ -0,0 +1,213 @@
+package operations
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
+	"github.com/spf13/cobra"
+
+	"github.com/algolia/cli/pkg/cmdutil"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+	"github.com/algolia/cli/pkg/prompt"
+	"github.com/algolia/cli/pkg/text"
+	"github.com/algolia/cli/pkg/utils"
+	"github.com/algolia/cli/pkg/validators"
+)
+
+type OperationsOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	SearchClient func() (*search.Client, error)
+
+	Wait bool
+
+	File    string
+	Scanner *bufio.Scanner
+
+	ContinueOnError bool
+}
+
+// NewOperationsCmd creates and returns an operations command for object operations
+func NewOperationsCmd(f *cmdutil.Factory, runF func(*OperationsOptions) error) *cobra.Command {
+	opts := &OperationsOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:               "operations -F <file> [--wait] [--continue-on-errors]",
+		Args:              validators.NoArgs(),
+		ValidArgsFunction: cmdutil.IndexNames(opts.SearchClient),
```

Comment:
I don't think you need a `ValidArgsFunction` since this command do not accept any args.